### ISSUE TITLE
MC 1.16.1  java.lang.NoSuchFieldError: PIG_ZOMBIE

### ISSUE THREAD

**Issue by Blueeyestar** [Shopkeepers/Shopkeepers#656]
Now the time has come. 1.16.1 has been released and gives all server owners happy sleepless nights. I hereby submit my modest contribution to it :)

Spigot-Version: CraftBukkit version git-Spigot-758abbe-38de0b8 (MC: 1.16.1) (Implementing API version 1.16.1-R0.1-SNAPSHOT)

[05:44:29] [Server thread/ERROR]: Could not load 'plugins\Shopkeepers-2.9.3.jar' in folder 'plugins'
org.bukkit.plugin.InvalidPluginException: 
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:139) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:394) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:301) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.craftbukkit.v1_16_R1.CraftServer.loadPlugins(CraftServer.java:377) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.DedicatedServer.init(DedicatedServer.java:186) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:808) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_231]
Caused by: java.lang.NoSuchFieldError: PIG_ZOMBIE
	at com.nisovin.shopkeepers.Settings.<clinit>(Settings.java:166) ~[?:?]
	at com.nisovin.shopkeepers.shopobjects.living.SKLivingShopObjectTypes.<init>(SKLivingShopObjectTypes.java:145) ~[?:?]
	at com.nisovin.shopkeepers.shopobjects.living.LivingShops.<init>(LivingShops.java:14) ~[?:?]
	at com.nisovin.shopkeepers.SKShopkeepersPlugin.<init>(SKShopkeepersPlugin.java:118) ~[?:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_231]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_231]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_231]
	at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_231]
	at java.lang.Class.newInstance(Unknown Source) ~[?:1.8.0_231]
	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:76) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:135) ~[minecraft_server.jar:git-Spigot-758abbe-38de0b8]
	... 7 more

**Comment by CloudeLecaw** [comment]
i can agree this
```
[07:55:22] [Server thread/ERROR]: Could not load 'plugins\Shopkeepers-2.9.3.jar' in folder 'plugins'
org.bukkit.plugin.InvalidPluginException: java.lang.NoSuchFieldError: PIG_ZOMBIE
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:139) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:394) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:301) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.craftbukkit.v1_16_R1.CraftServer.loadPlugins(CraftServer.java:377) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.DedicatedServer.init(DedicatedServer.java:186) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:808) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: java.lang.NoSuchFieldError: PIG_ZOMBIE
	at com.nisovin.shopkeepers.Settings.<clinit>(Settings.java:166) ~[?:?]
	at com.nisovin.shopkeepers.shopobjects.living.SKLivingShopObjectTypes.<init>(SKLivingShopObjectTypes.java:145) ~[?:?]
	at com.nisovin.shopkeepers.shopobjects.living.LivingShops.<init>(LivingShops.java:14) ~[?:?]
	at com.nisovin.shopkeepers.SKShopkeepersPlugin.<init>(SKShopkeepersPlugin.java:118) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[?:?]
	at java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:124) ~[?:?]
	at jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:346) ~[?:?]
	at java.lang.Class.newInstance(Class.java:604) ~[?:?]
	at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:76) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:135) ~[spigot-1.16.1.jar:git-Spigot-758abbe-38de0b8]
	... 7 more
```

**Comment by PulseBeat02** [comment]
Same here, I got the exact same error. Seems that other plugins and datapacks are now messed up because of the removal of the Zombie Pigman in 1.16.1.

**Comment by Blueeyestar** [comment]
The problem can be solved by replacing the following list in the config:

enabled-living-shops:
- BAT
- BEE
- BLAZE
- CAT
- CAVE_SPIDER
- CHICKEN
- COD
- COW
- CREEPER
- DOLPHIN
- DONKEY
- DROWNED
- ELDER_GUARDIAN
- ENDER_DRAGON
- ENDERMAN
- ENDERMITE
- EVOKER
- FOX
- GHAST
- GIANT
- GUARDIAN
- HOGLIN
- HORSE
- HUSK
- ILLUSIONER
- IRON_GOLEM
- LLAMA
- MAGMA_CUBE
- MULE
- MUSHROOM_COW
- OCELOT
- PANDA
- PARROT
- PHANTOM
- PIG
- PIGLIN
- PILLAGER
- POLAR_BEAR
- PUFFERFISH
- RABBIT
- RAVAGER
- SALMON
- SHEEP
- SHULKER
- SILVERFISH
- SKELETON
- SKELETON_HORSE
- SLIME
- SNOWMAN
- SPIDER
- SQUID
- STRAY
- STRIDER
- TRADER_LLAMA
- TROPICAL_FISH
- TURTLE
- VEX
- VILLAGER
- VINDICATOR
- WANDERING_TRADER
- WITCH
- WITHER
- WITHER_SKELETON
- WOLF
- ZOGLIN
- ZOMBIE
- ZOMBIE_HORSE
- ZOMBIE_VILLAGER
- ZOMBIFIED_PIGLIN

**Comment by adelando** [comment]
Nope, changing that in the config still doesnt work.

**Comment by blablubbabc** [comment]
This error is resolved by using the latest version of Spigot 1.16.1.

However, this alone doesn't make Shopkeepers safe to be used on MC 1.16.x yet. An update is one the way.

**Comment by blablubbabc** [comment]
v2.10.0 has been uploaded and includes the update for MC 1.16.1.
